src/app/stack/stack-details/stack-details.component.ts

```diff
@@ -257,43 +250,42 @@ export class StackDetailsComponent implements OnChanges {
-                    if (error.status >= 500) {
-                        title = 'Something unexpected happened. Please try again later.';
-                        this.tokenAlertsMessage = 'Something unexpected happened. Please try again later.'
-                        this.tokenAlertType = 'red';
-                    } else if (error.status === 400) {
-                        title = 'Invalid Token';
-                        this.tokenAlertsMessage = 'Invalid Token'
-                        this.tokenAlertType = 'red';
-                    } else if (error.status === 401) {
-                        title =
-                            'You don\'t seem to have sufficient privileges to access this - Request unauthorized';
-                        this.tokenAlertsMessage = 'Request unauthorized'
-                        this.tokenAlertType = 'red';
-                    } else {
-                        title = 'Error in Submit Token. Please try again later.';
-                        this.tokenAlertsMessage = 'Error in Submit Token. Please try again later.'
-                        this.tokenAlertType = 'red';
-                    }
+
+        this.loaderStatus = true;
+        await this.stackAnalysisService.linkSynkTokenWithUserID(this.getBaseUrl(this.stack), this.uuid, this.token, this.gatewayConfig)
+            .then(res => {
+                if (res.status === 200) {
+                    this.init();
                     this.loaderStatus = false;
-                    console.log(title);
-                });
-        }
+                    this.token = '';
+                    setTimeout(() => {
+                        this.modalRef.hide();
+                    }, (1000));
+                }
+            })
+            .catch(error => {
+                let title: string = '';
+                if (error.status >= 500) {
+                    title = 'Something unexpected happened. Please try again later.';
+                    this.tokenAlertsMessage = 'Something unexpected happened. Please try again later.'
+                    this.tokenAlertType = 'red';
+                } else if (error.status === 400) {
+                    title = 'Invalid Token';
+                    this.tokenAlertsMessage = 'Invalid Token'
+                    this.tokenAlertType = 'red';
+                } else if (error.status === 401) {
+                    title =
+                        'You don\'t seem to have sufficient privileges to access this - Request unauthorized';
+                    this.tokenAlertsMessage = 'Request unauthorized'
+                    this.tokenAlertType = 'red';
+                } else {
+                    title = 'Error in Submit Token. Please try again later.';
+                    this.tokenAlertsMessage = 'Error in Submit Token. Please try again later.'
+                    this.tokenAlertType = 'red';
+                }
+                this.loaderStatus = false;
+                console.log(title);
```

Comment:
title has been logged twice is this necessary because we are showing the same message in the alert

Reply:
sorry got it

Reply:
LGTM

Reply:
ya I missed to remove it, will remove them, none are required.

Reply:
It is fixed in 168 PR